sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -372,31 +368,52 @@ private HttpRequest configRequest(final HttpRequest request, final SwaggerMethod
-        try {
-            final Constructor<? extends HttpResponseException> exceptionConstructor = exception.getExceptionType()
-                .getConstructor(String.class, HttpResponse.class, exception.getExceptionBodyType());
-            result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + bodyRepresentation,
-                httpResponse, responseDecodedContent);
-        } catch (ReflectiveOperationException e) {
-            String message = "Status code " + responseStatusCode + ", but an instance of "
-                + exception.getExceptionType().getCanonicalName() + " cannot be created."
-                + " Response body: " + bodyRepresentation;
-
-            result = new IOException(message, e);
+        // For HttpResponseException types that exist in azure-core, call the constructor directly.
+        Class<? extends HttpResponseException> exceptionType = exception.getExceptionType();
+        if (exceptionType == HttpResponseException.class) {
+            return new HttpResponseException(exceptionMessage.toString(), httpResponse, responseDecodedContent);
+        } else if (exceptionType == ClientAuthenticationException.class) {
+            return new ClientAuthenticationException(exceptionMessage.toString(), httpResponse, responseDecodedContent);
+        } else if (exceptionType == DecodeException.class) {
+            return new DecodeException(exceptionMessage.toString(), httpResponse, responseDecodedContent);
+        } else if (exceptionType == ResourceExistsException.class) {
+            return new ResourceExistsException(exceptionMessage.toString(), httpResponse, responseDecodedContent);
+        } else if (exceptionType == ResourceModifiedException.class) {
+            return new ResourceModifiedException(exceptionMessage.toString(), httpResponse, responseDecodedContent);
+        } else if (exceptionType == ResourceNotFoundException.class) {
+            return new ResourceNotFoundException(exceptionMessage.toString(), httpResponse, responseDecodedContent);
+        } else if (exceptionType == TooManyRedirectsException.class) {
+            return new TooManyRedirectsException(exceptionMessage.toString(), httpResponse, responseDecodedContent);
+        } else {
+            // Finally, if the HttpResponseException subclass doesn't exist in azure-core, use reflection to create a
+            // new instance of it.
+            try {
+                MethodHandle handle = RESPONSE_EXCEPTION_CONSTRUCTOR_CACHE.get(exceptionType,
+                    exception.getExceptionBodyType());
+                return RESPONSE_EXCEPTION_CONSTRUCTOR_CACHE.invoke(handle, exceptionMessage.toString(), httpResponse,
+                    responseDecodedContent);
+            } catch (RuntimeException e) {
+                // And if reflection fails, return an IOException.
+                // TODO (alzimmer): Determine if this should be an IOException or HttpResponseException.
```

Comment:
I think neither, if we can't locate and use ctor that's neither IO nor response fault. IllegalStateException maybe?

Reply:
For now, since it's already an IOException I'm going to leave it as an IOException